### ISSUE TITLE
patternplate-client throws "Cannot add a pure render decorator to Icon"

### ISSUE THREAD

**Issue by marionebl** [sinnerschrader/patternplate#109]
### Actual
`patternplate-client` throws an exception during start

### Expected
`patternplate-client` starts without throwing

### Affected versions
|Software | Version |
|---|---|
| **patternplate** | v0.15.16 - v0.18.1 |
| **patternplate-client** |  v0.15.0 - v0.16.2 |

### Details

```
Error: Cannot add a pure render decorator to Icon, because it already implements `shouldComponentUpdate()`.
    at pureRenderDecorator (~~/patternplate-client/node_modules/pure-render-decorator/index.js:44:11)
    at Object.<anonymous> (~~/patternplate-client/application/components/common/icon.js:155:46)
```

<details>
<summary>Full error log</summary>

```
❯ npm start

> patternplate-client@0.16.2 start /Users/marneb/Projects/patternplate/test
> node --harmony binary/patternplate-client.js

2016-10-31T22:42:21.603Z - silly: [patternplate-client] patternplate-client Registering hook 'configure'
2016-10-31T22:42:21.605Z - silly: [patternplate-client] patternplate-client Registering hook 'console'
2016-10-31T22:42:21.605Z - silly: [patternplate-client] patternplate-client Registering hook 'engine'
2016-10-31T22:42:21.605Z - silly: [patternplate-client] patternplate-client Registering hook 'log'
2016-10-31T22:42:21.606Z - silly: [patternplate-client] patternplate-client Registering hook 'middlewares'
2016-10-31T22:42:21.606Z - silly: [patternplate-client] patternplate-client Registering hook 'routes'
2016-10-31T22:42:21.606Z - silly: [patternplate-client] patternplate-client Registering hook 'user-hooks'
2016-10-31T22:42:21.606Z - debug: [patternplate-client] patternplate-client Running stage 'configure' on hook 'configure'
2016-10-31T22:42:21.606Z - debug: [patternplate-client] patternplate-client Ran stage 'configure' on hook 'configure'
2016-10-31T22:42:21.606Z - debug: [patternplate-client] patternplate-client Running stage 'start' on hook 'configure'
2016-10-31T22:42:21.607Z - info: [patternplate-client] patternplate-client Loading configuration from '~~/patternplate-client/configuration'
2016-10-31T22:42:21.607Z - debug: [patternplate-client] patternplate-client Ran stage 'start' on hook 'configure'
2016-10-31T22:42:21.607Z - debug: [patternplate-client] patternplate-client Running stage 'configure' on hook 'user-hooks'
2016-10-31T22:42:21.607Z - debug: [patternplate-client] patternplate-client Ran stage 'configure' on hook 'user-hooks'
2016-10-31T22:42:21.607Z - debug: [patternplate-client] patternplate-client Running stage 'start' on hook 'user-hooks'
2016-10-31T22:42:21.607Z - silly: [patternplate-client] patternplate-client Registering hook 'configure'
2016-10-31T22:42:21.607Z - silly: [patternplate-client] patternplate-client Registering hook 'console'
2016-10-31T22:42:21.607Z - silly: [patternplate-client] patternplate-client Registering hook 'engine'
2016-10-31T22:42:21.608Z - silly: [patternplate-client] patternplate-client Registering hook 'log'
2016-10-31T22:42:21.608Z - silly: [patternplate-client] patternplate-client Registering hook 'middlewares'
2016-10-31T22:42:21.608Z - silly: [patternplate-client] patternplate-client Registering hook 'routes'
2016-10-31T22:42:21.608Z - silly: [patternplate-client] patternplate-client Registering hook 'user-hooks'
2016-10-31T22:42:21.608Z - debug: [patternplate-client] patternplate-client Running stage 'configure' on hook 'log'
2016-10-31T22:42:21.608Z - debug: [patternplate-client] patternplate-client Ran stage 'configure' on hook 'log'
2016-10-31T22:42:21.608Z - debug: [patternplate-client] patternplate-client Running stage 'start' on hook 'log'
2016-10-31T22:42:21.608Z - silly: [patternplate-client] patternplate-client Draining boot logger queue...
2016-10-31T22:42:21.609Z - silly: [patternplate-client] Deploying application logger...
2016-10-31T22:42:21.609Z - debug: [patternplate-client] Ran stage 'start' on hook 'log'
2016-10-31T22:42:21.609Z - debug: [patternplate-client] Running stage 'configure' on hook 'console'
2016-10-31T22:42:21.609Z - debug: [patternplate-client] Running stage 'configure' on hook 'engine'
2016-10-31T22:42:21.610Z - debug: [patternplate-client] Ran stage 'configure' on hook 'console'
2016-10-31T22:42:21.610Z - debug: [patternplate-client] Ran stage 'configure' on hook 'engine'
2016-10-31T22:42:21.610Z - debug: [patternplate-client] Running stage 'start' on hook 'console'
2016-10-31T22:42:21.610Z - debug: [patternplate-client] Running stage 'start' on hook 'engine'
2016-10-31T22:42:21.612Z - debug: [patternplate-client] Ran stage 'start' on hook 'engine'
2016-10-31T22:42:21.612Z - debug: [patternplate-client] Running stage 'configure' on hook 'routes'
2016-10-31T22:42:21.613Z - debug: [patternplate-client] Ran stage 'configure' on hook 'routes'
2016-10-31T22:42:21.613Z - debug: [patternplate-client] Running stage 'start' on hook 'routes'
2016-10-31T22:42:21.632Z - debug: [patternplate-client] Ran stage 'start' on hook 'console'
2016-10-31T22:42:22.098Z - error: [patternplate-client] An error ocurred on stage start of hook 'routes'
2016-10-31T22:42:22.098Z - error: [patternplate-client] Error: Cannot add a pure render decorator to Icon, because it already implements `shouldComponentUpdate()`.
    at pureRenderDecorator (~~/patternplate-client/node_modules/pure-render-decorator/index.js:44:11)
    at Object.<anonymous> (~~/patternplate-client/application/components/common/icon.js:155:46)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (~~/patternplate-client/application/components/navigation/toolbar.js:55:13)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (~~/patternplate-client/application/components/index.js:81:16)
2016-10-31T22:42:22.099Z - error: [patternplate-client] An error ocurred on stage start of hook 'user-hooks'
2016-10-31T22:42:22.099Z - error: [patternplate-client] Error: Cannot add a pure render decorator to Icon, because it already implements `shouldComponentUpdate()`.
    at pureRenderDecorator (~~/patternplate-client/node_modules/pure-render-decorator/index.js:44:11)
    at Object.<anonymous> (~~/patternplate-client/application/components/common/icon.js:155:46)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (~~/patternplate-client/application/components/navigation/toolbar.js:55:13)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (~~/patternplate-client/application/components/index.js:81:16)
2016-10-31T22:42:22.099Z - error: [patternplate-client] Error: Cannot add a pure render decorator to Icon, because it already implements `shouldComponentUpdate()`.
    at pureRenderDecorator (~~/patternplate-client/node_modules/pure-render-decorator/index.js:44:11)
    at Object.<anonymous> (~~/patternplate-client/application/components/common/icon.js:155:46)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (~~/patternplate-client/application/components/navigation/toolbar.js:55:13)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (~~/patternplate-client/application/components/index.js:81:16)
silly patternplate-client Registering hook 'configure'
silly patternplate-client Registering hook 'console'
silly patternplate-client Registering hook 'engine'
silly patternplate-client Registering hook 'log'
silly patternplate-client Registering hook 'middlewares'
silly patternplate-client Registering hook 'routes'
silly patternplate-client Registering hook 'user-hooks'
debug patternplate-client Running stage 'configure' on hook 'configure'
debug patternplate-client Ran stage 'configure' on hook 'configure'
debug patternplate-client Running stage 'start' on hook 'configure'
info patternplate-client Loading configuration from '~~/patternplate-client/configuration'
debug patternplate-client Ran stage 'start' on hook 'configure'
debug patternplate-client Running stage 'configure' on hook 'user-hooks'
debug patternplate-client Ran stage 'configure' on hook 'user-hooks'
debug patternplate-client Running stage 'start' on hook 'user-hooks'
silly patternplate-client Registering hook 'configure'
silly patternplate-client Registering hook 'console'
silly patternplate-client Registering hook 'engine'
silly patternplate-client Registering hook 'log'
silly patternplate-client Registering hook 'middlewares'
silly patternplate-client Registering hook 'routes'
silly patternplate-client Registering hook 'user-hooks'
debug patternplate-client Running stage 'configure' on hook 'log'
debug patternplate-client Ran stage 'configure' on hook 'log'
debug patternplate-client Running stage 'start' on hook 'log'
silly patternplate-client Draining boot logger queue...
Trace: [Error: Cannot add a pure render decorator to Icon, because it already implements `shouldComponentUpdate()`.]
    at _callee$ (~~/patternplate-client/binary/patternplate-client.js:34:15)
    at tryCatch (~~/patternplate-client/node_modules/babel-runtime/node_modules/regenerator-runtime/runtime.js:62:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (~~/patternplate-client/node_modules/babel-runtime/node_modules/regenerator-runtime/runtime.js:336:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as throw] (~~/patternplate-client/node_modules/babel-runtime/node_modules/regenerator-runtime/runtime.js:95:21)
    at step (~~/patternplate-client/node_modules/babel-runtime/helpers/asyncToGenerator.js:17:30)
    at ~~/patternplate-client/node_modules/babel-runtime/helpers/asyncToGenerator.js:30:13
    at run (~~/patternplate-client/node_modules/babel-runtime/node_modules/core-js/library/modules/es6.promise.js:87:22)
    at ~~/patternplate-client/node_modules/babel-runtime/node_modules/core-js/library/modules/es6.promise.js:100:28
    at flush (~~/patternplate-client/node_modules/babel-runtime/node_modules/core-js/library/modules/_microtask.js:18:9)
    at nextTickCallbackWith0Args (node.js:420:9)
    at process._tickCallback (node.js:349:13)
~~/patternplate-client/binary/patternplate-client.js:82
		throw err;
		^

Error: Error: Cannot add a pure render decorator to Icon, because it already implements `shouldComponentUpdate()`.
    at _callee$ (~~/patternplate-client/binary/patternplate-client.js:35:13)
    at tryCatch (~~/patternplate-client/node_modules/babel-runtime/node_modules/regenerator-runtime/runtime.js:62:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (~~/patternplate-client/node_modules/babel-runtime/node_modules/regenerator-runtime/runtime.js:336:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as throw] (~~/patternplate-client/node_modules/babel-runtime/node_modules/regenerator-runtime/runtime.js:95:21)
    at step (~~/patternplate-client/node_modules/babel-runtime/helpers/asyncToGenerator.js:17:30)
    at ~~/patternplate-client/node_modules/babel-runtime/helpers/asyncToGenerator.js:30:13
    at run (~~/patternplate-client/node_modules/babel-runtime/node_modules/core-js/library/modules/es6.promise.js:87:22)
    at ~~/patternplate-client/node_modules/babel-runtime/node_modules/core-js/library/modules/es6.promise.js:100:28
    at flush (~~/patternplate-client/node_modules/babel-runtime/node_modules/core-js/library/modules/_microtask.js:18:9)
    at nextTickCallbackWith0Args (node.js:420:9)
    at process._tickCallback (node.js:349:13)

npm ERR! Darwin 15.6.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "start"
npm ERR! node v4.4.6
npm ERR! npm  v2.15.5
npm ERR! code ELIFECYCLE
npm ERR! patternplate-client@0.16.2 start: `node --harmony binary/patternplate-client.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the patternplate-client@0.16.2 start script 'node --harmony binary/patternplate-client.js'.
npm ERR! This is most likely a problem with the patternplate-client package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node --harmony binary/patternplate-client.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs patternplate-client
npm ERR! Or if that isn't available, you can get their info via:
npm ERR! 
npm ERR!     npm owner ls patternplate-client
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     ~~/patternplate-client/npm-debug.log
```
</details>

### Steps to reproduce

```shell
git clone git@github.com:sinnerschrader/patternplate-client.git
cd patternplate-client
git checkout v0.16.2 # => HEAD is now at 56275e7... 0.16.2
npm install
npm run build
npm start
```

## Analysis
This is caused by [pure-render-decorator](https://github.com/felixgirault/pure-render-decorator/pull/8) introducing a breaking change without marking the fact with a major version bump. 
It now throws when applied to components declaring a `shouldComponentUpdate` method. 

This is causing `patternplate-client` to fail during CommonJS module evaluation, thus affecting `patternplate` and `patternplate-console` commands, too.

**Comment by marionebl** [comment]
## Published versions

https://github.com/sinnerschrader/patternplate-client/releases/tag/v0.15.4
https://github.com/sinnerschrader/patternplate-client/releases/tag/v0.16.3

Everything `>= 1.0.0` was not affected as the `<Icon />` implementation changed to omit the empty `shouldComponentUpdate` method.

**Comment by felixgirault** [comment]
Hi,

Sorry for the mess on `pure-render-decorator`...
Just so you know, I just released version 1.2.1, which should fix the problem by replacing errors with warnings ;)

**Comment by marionebl** [comment]
No worries @felixgirault! Thanks for releasing the patched version, will use that for the next version of patternplate.
